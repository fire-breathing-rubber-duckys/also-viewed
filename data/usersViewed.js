const usersViewed = [
  {
    productId: 1,
    photoURL: 'https://www.rei.com/media/product/148949?size=280',
    brandName: 'Manduka',
    itemName: 'Lean Cork Block',
    itemRating: 5,
    itemPrice: '$16.00',
    ratingCount: 2,
  },
  {
    productId: 2,
    photoURL: 'https://www.rei.com/media/product/182895?size=280',
    brandName: 'Manduka',
    itemName: 'Rectangular Bolster',
    itemRating: 3,
    itemPrice: '$68.00',
    ratingCount: 2,
  },
  {
    productId: 3,
    photoURL: 'https://www.rei.com/media/product/173243?size=280',
    brandName: 'Manduka',
    itemName: 'unBLOK Recycled Foam Yoga Block',
    itemRating: 5,
    itemPrice: '$24.00',
    ratingCount: 3,
  },
  {
    productId: 4,
    photoURL: 'https://www.rei.com/media/product/101409?size=280',
    brandName: 'Jade',
    itemName: 'Cork Yoga Block',
    itemRating: 4.5,
    itemPrice: '$14.95-$19.95',
    ratingCount: 6,
  },
  {
    productId: 5,
    photoURL: 'https://www.rei.com/media/product/182894?size=280',
    brandName: 'Manduka',
    itemName: 'Lean Bolster',
    itemRating: 2,
    itemPrice: '$43.95 $56.00',
    ratingCount: 1,
  },
  {
    productId: 6,
    photoURL: 'https://www.rei.com/media/product/843154?size=280',
    brandName: 'Manduka',
    itemName: 'Recycled Foam Yoga Block',
    itemRating: 5,
    itemPrice: '$16.00',
    ratingCount: 8,
  },
  {
    productId: 7,
    photoURL: 'https://www.rei.com/media/product/850808?size=280',
    brandName: 'Manduka',
    itemName: 'PRO Yoga Mat - Black Sage',
    itemRating: 5,
    itemPrice: '$120.00',
    ratingCount: 18,
  },
  {
    productId: 8,
    photoURL: 'https://www.rei.com/media/product/112757?size=280',
    brandName: 'Manduka',
    itemName: 'Go Move Yoga Mat Sling',
    itemRating: 3,
    itemPrice: '$20.00',
    ratingCount: 77,
  },
  {
    productId: 9,
    photoURL: 'https://www.rei.com/media/product/786833?size=280',
    brandName: 'Jade',
    itemName: 'Harmony Professional Yoga Mat',
    itemRating: 4,
    itemPrice: '$79.95',
    ratingCount: 48,
  },
  {
    productId: 10,
    photoURL: 'https://www.rei.com/media/product/860616?size=280',
    brandName: 'Pro-Tec Athletics',
    itemName: 'Exercise Stretch Band',
    itemRating: 0,
    itemPrice: '$24.95 $29.99',
    ratingCount: 0,
  },
  {
    productId: 11,
    photoURL: 'https://www.rei.com/media/product/120819?size=280',
    brandName: 'Manduka',
    itemName: 'X Yoga Mat',
    itemRating: 4.5,
    itemPrice: '$58.00',
    ratingCount: 7,
  },
];

module.exports = usersViewed;
