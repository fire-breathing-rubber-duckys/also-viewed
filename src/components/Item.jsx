/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import StarRating from './StarRating';

const ItemWrapper = styled.div`
  margin-left: 15px;
  margin-right: 15px;
  margin-bottom: 55px;
  position: relative;
  z-index: 1;
`;

const Item = ({ item }) => {
  const stringArr = item.itemName.split(' ');
  const ratingCount = ` (${item.ratingCount})`;
  const ip = item.itemPrice;
  const stringOne = [];
  const stringTwo = [];

  // Logic to filter item price by reg Vs sale item //
  let price;
  if (ip.length < 8 || ip[6] !== ' ') {
    price = <label className="item-price">{ip}</label>;
  } else {
    const priceArr = ip.split(' ');
    const priceOne = ` ${priceArr[0]}`;
    const priceTwo = ` ${priceArr[1]}`;
    price = (
      <div>
        <label className="item-sale-discount">{priceOne}</label>
        <label className="item-sale-price">{priceTwo}</label>
      </div>
    );
  }

  // Logic to filter the itemName to format in a nice fashion //
  if (stringArr[0].length > 8 && stringArr.length === 2) {
    stringOne.push(stringArr[0]);
    stringTwo.push(stringArr[1]);
  } else if (stringArr[0].length > 8 && stringArr.length === 3) {
    stringOne.push(stringArr[0]);
    stringTwo.push(stringArr[1], stringArr[2]);
  } else {
    for (let i = 0; i < stringArr.length; i += 1) {
      if (i < 2) {
        stringOne.push(stringArr[i]);
      } else {
        stringTwo.push(stringArr[i]);
      }
    }
  }

  return (
    <>
      <ItemWrapper>
        <img src={item.photoURL} alt="" className="also-viewed-img" />
        <div>
          <div className="item-name">
            <div className="item-brand">{item.brandName}</div>
            {stringOne.join(' ')}
            <br />
            {stringTwo.join(' ')}
          </div>
          <StarRating rating={item.itemRating} ratingCount={ratingCount} />
          {price}
        </div>
      </ItemWrapper>
    </>
  );
};

Item.propTypes = {
  item: PropTypes.instanceOf(Object).isRequired,
};

export default Item;
