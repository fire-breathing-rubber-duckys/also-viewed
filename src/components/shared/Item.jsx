import React from 'react';
import PropTypes from 'prop-types';

const Item = ({ item }) => (
  <div className="flex-item">
    <img src={item.photoURL} alt="" />
    <div>{item.brandName}</div>
    <div>{item.itemName}</div>
    <div>{item.itemRating}</div>
    <div>{item.ratingCount}</div>
    <div>{item.itemPrice}</div>
  </div>
);

Item.propTypes = {
  item: PropTypes.instanceOf(Object).isRequired,
};

export default Item;
