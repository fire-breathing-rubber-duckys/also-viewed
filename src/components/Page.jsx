import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Item from './Item';

const Flex = styled.div`
  display: flex;
`;

const Page = ({ data, page }) => (
  <>
    <Flex>
      {data.map((item) => {
        const valid = page === 0 ? [1, 2, 3, 4, 5, 6] : [6, 7, 8, 9, 10, 11];
        if (valid.includes(item.productId)) {
          return <Item item={item} key={item.productId} />;
        }
      })}
    </Flex>
  </>
);

Page.propTypes = {
  data: PropTypes.instanceOf(Array).isRequired,
  page: PropTypes.number.isRequired,
};

export default Page;
