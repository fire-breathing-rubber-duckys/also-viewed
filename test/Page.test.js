import React from 'react';
import { shallow } from 'enzyme';
import { expect as expectChai } from 'chai';
import renderer from 'react-test-renderer';
import Item from '../src/components/Item';
import usersViewed from '../data/usersViewed';
import Page from '../src/components/Page';

const dataTwo = usersViewed.slice(5, usersViewed.length);
const dataOne = usersViewed.slice(0, 6);
const page = 0;

it('Should render correctly', () => {
  const tree = renderer
    .create(<Page data={dataOne} page={page} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Should render six items', () => {
  const wrapper = shallow(<Page data={dataOne} page={page} />);
  expectChai(wrapper.find(Item)).to.be.lengthOf(6);
});