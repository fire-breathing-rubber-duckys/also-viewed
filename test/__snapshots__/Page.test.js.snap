// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div
  className="sc-gsTCUz ywWHk"
>
  <div
    className="sc-bdfBwQ fECFQP"
  >
    <img
      alt=""
      className="also-viewed-img"
      src="https://www.rei.com/media/product/148949?size=280"
    />
    <div>
      <div
        className="item-name"
      >
        <div
          className="item-brand"
        >
          Manduka
        </div>
        Lean Cork
        <br />
        Block
      </div>
      <div>
        <label
          className="item-rating-count"
        >
           (2)
        </label>
      </div>
      <label
        className="item-price"
      >
        $16.00
      </label>
    </div>
  </div>
  <div
    className="sc-bdfBwQ fECFQP"
  >
    <img
      alt=""
      className="also-viewed-img"
      src="https://www.rei.com/media/product/182895?size=280"
    />
    <div>
      <div
        className="item-name"
      >
        <div
          className="item-brand"
        >
          Manduka
        </div>
        Rectangular
        <br />
        Bolster
      </div>
      <div>
        <label
          className="item-rating-count"
        >
           (2)
        </label>
      </div>
      <label
        className="item-price"
      >
        $68.00
      </label>
    </div>
  </div>
  <div
    className="sc-bdfBwQ fECFQP"
  >
    <img
      alt=""
      className="also-viewed-img"
      src="https://www.rei.com/media/product/173243?size=280"
    />
    <div>
      <div
        className="item-name"
      >
        <div
          className="item-brand"
        >
          Manduka
        </div>
        unBLOK Recycled
        <br />
        Foam Yoga Block
      </div>
      <div>
        <label
          className="item-rating-count"
        >
           (3)
        </label>
      </div>
      <label
        className="item-price"
      >
        $24.00
      </label>
    </div>
  </div>
  <div
    className="sc-bdfBwQ fECFQP"
  >
    <img
      alt=""
      className="also-viewed-img"
      src="https://www.rei.com/media/product/101409?size=280"
    />
    <div>
      <div
        className="item-name"
      >
        <div
          className="item-brand"
        >
          Jade
        </div>
        Cork Yoga
        <br />
        Block
      </div>
      <div>
        <label
          className="item-rating-count"
        >
           (6)
        </label>
      </div>
      <label
        className="item-price"
      >
        $14.95-$19.95
      </label>
    </div>
  </div>
  <div
    className="sc-bdfBwQ fECFQP"
  >
    <img
      alt=""
      className="also-viewed-img"
      src="https://www.rei.com/media/product/182894?size=280"
    />
    <div>
      <div
        className="item-name"
      >
        <div
          className="item-brand"
        >
          Manduka
        </div>
        Lean Bolster
        <br />
        
      </div>
      <div>
        <label
          className="item-rating-count"
        >
           (1)
        </label>
      </div>
      <div>
        <label
          className="item-sale-discount"
        >
           $43.95
        </label>
        <label
          className="item-sale-price"
        >
           $56.00
        </label>
      </div>
    </div>
  </div>
  <div
    className="sc-bdfBwQ fECFQP"
  >
    <img
      alt=""
      className="also-viewed-img"
      src="https://www.rei.com/media/product/843154?size=280"
    />
    <div>
      <div
        className="item-name"
      >
        <div
          className="item-brand"
        >
          Manduka
        </div>
        Recycled Foam
        <br />
        Yoga Block
      </div>
      <div>
        <label
          className="item-rating-count"
        >
           (8)
        </label>
      </div>
      <label
        className="item-price"
      >
        $16.00
      </label>
    </div>
  </div>
</div>
`;
